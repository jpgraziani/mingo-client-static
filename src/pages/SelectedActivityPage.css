.single-activity {
  background-color: var(--lightBlue);
  color: var(--primary);
  width: 100%;
  padding-bottom: 2rem;
}

.single-activity-info {
  width: 80vw;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  padding: 2rem 0;
}

.activity-title {
  width: 100%;
  padding-left: 1rem;
}


.directions, .supplies {
  white-space: pre-line;
  flex: 1;
  padding-top: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.activity-title h2 {
  margin-bottom: .25rem;
}

.activity-title h2, .directions h3, .supplies h3 {
  color: var(--primary);
  text-transform: capitalize;
  letter-spacing: var(--mainSpacing);
}

.activity-title h3 {
  margin-top: 0;
}

.supplies {
  width: 80vw;
  margin: 0 auto 3rem auto;
}

.hr {
  width: 100%;
  height: 3px;
  background-color: var(--primary);
}

.lower {
  height: 1.5px;
  margin-bottom: 1rem;
}

.single-activity .back-btn {
  margin-right: 1rem;
  display: inline-block;
	padding: .9rem 1rem;
	text-decoration: none;
	font-size: 1rem;
	letter-spacing: 1px;
	font-weight: 600;
  border: 2px solid var(--white);
  background: none;
  color: var(--white);
}

.single-activity .back-btn:hover {
  border: 2px solid white;
  background: white;
  color: var(--primary);
}

.back-container {
  width: 100%;
  padding: 1rem;
}

@media (min-width:769px) {
	.single-activity-info {
		flex-direction: row;
    flex-wrap: wrap;
	}
}