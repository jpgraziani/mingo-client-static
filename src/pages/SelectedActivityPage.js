import React from 'react'

import { findActivity } from '../ActivityHelpers'
import ApiContext from '../ApiContext'

export default class SelectedActivityPage extends React.Component {
  render

  static contextType = ApiContext

  render() {
    const { activities = [] } = this.context
    const { activityId } = this.props.match.params
    const activity = findActivity(activities, activityId) || { content: ''}

    return (
      <>
        <article className='single-activity'>
          <div className='single-activity-info'>
            <section>
              <h3>{activity.name}</h3>
            </section>
            <section className='desc'>
              <hr />
              <h3>Supplies Needed</h3>
              {activity.supplies}
            </section>
            <section>
              <hr />
              <h3>About Activity</h3>
              {activity.description}
            </section>
          </div>
        </article>
      </>
    );
  }
}