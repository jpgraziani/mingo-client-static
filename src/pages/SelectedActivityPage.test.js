import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SelectedActivityPage from './SelectedActivityPage'

describe(`SelectedActivityPage component`, () => {
  it('renders a SelectedActivityPage by default', () => {
    const wrapper = shallow(<SelectedActivityPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

