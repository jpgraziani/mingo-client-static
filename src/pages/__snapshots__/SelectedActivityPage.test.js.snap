// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedActivityPage component renders a SelectedActivityPage by default 1`] = `
<article
  className="single-activity"
>
  <div
    className="single-activity-info"
  >
    <section
      className="activity-title"
    >
      <h2 />
    </section>
    <div
      className="back-container"
    >
      <Link
        className="back-btn"
        to="/activities"
      >
        previous
      </Link>
      <Link
        className="back-btn"
        to="/add-activity"
      >
        add activity
      </Link>
    </div>
    <section
      className="supplies"
    >
      <div
        className="hr"
      />
      <h3>
        Supplies
      </h3>
      <div
        className="hr lower"
      />
      <p />
    </section>
    <section
      className="directions"
    >
      <div
        className="hr"
      />
      <h3>
        About Activity
      </h3>
      <div
        className="hr lower"
      />
      <p />
    </section>
  </div>
</article>
`;
